### PR TITLE
Istio prow: upgrade prometheus operator from 0.29.0 to 0.45.0

### PR DESCRIPTION
**Pull request by chaodaiG**
This was based on https://github.com/kubernetes/test-infra/pull/21291

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by chaodaiG** [comment]
Fixed a lint error, need another lgtm

File: prow/Makefile [hunk 1]
```diff
@@ -40,6 +40,7 @@ deploy-velodrome: get-cluster-credentials get-api-resources
 # TODO(clarketm): enable pruning
 deploy-monitoring: get-cluster-credentials
 	$(MAKE) -C cluster/monitoring/mixins clean apply
+	kubectl apply -f ./cluster/monitoring/prometheus-operator-crds/ -n prow-monitoring
```
Comment (cjwagner) [inline prow/Makefile:43]: FYI CRDs are cluster scoped so `-n prow-monitoring` doesn't make a difference here. Shouldn't hurt though.

**Review by cjwagner — APPROVED** [review]
I just deleted the old CRDs from the cluster.

**Pull request merged**